### PR TITLE
Swaggerhub payment api 2.3.0

### PR DESCRIPTION
**Pull request by amruta-09**
## Check yourself

- [ ] Code is linted
- [ ] Tested
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: openapi/paymentApi.yaml [hunk 1]
```diff
@@ -2021,67 +2021,91 @@
...
             "type" : "string",
-            "description" : "Payment currency"
+            "description" : "Payment currency. The valid country currency code should be per ISO 4217 standard.",
+            "example" : "USD"
           },
-          "minPaymentAmount" : {
+          "minAmount" : {
             "type" : "number",
-            "description" : "Minimum payment amount"
+            "description" : "Minimum payment amount.",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2048]: 1. Provide what is the possible minimum amount based on the speed.

File: openapi/paymentApi.yaml [hunk 2]
```diff
@@ -2021,67 +2021,91 @@
...
+            "example" : 100.0
           },
-          "maxPaymentAmount" : {
+          "maxAmount" : {
             "type" : "number",
-            "description" : "Maximum payment amount"
+            "description" : "Maximum payment amount.",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2053]: 1. Provide what is the possible maximum amount based on the speed.

File: openapi/paymentApi.yaml [hunk 3]
```diff
@@ -2021,67 +2021,91 @@
...
+            "example" : 1000.0
           },
-          "forexContractNo" : {
-            "type" : "string",
-            "description" : "FX contract number"
+          "paymentAmount" : {
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2056]: Q: What is the difference between paymentAmount, minAmount and maxAmount?

File: openapi/paymentApi.yaml [hunk 4]
```diff
@@ -2021,67 +2021,91 @@
...
           "senderName" : {
             "type" : "string",
-            "description" : "Sender name associated with the payment."
+            "description" : "Sender name associated with the payment.",
+            "example" : "corp"
           },
           "senderAccountNo" : {
             "type" : "string",
-            "description" : "Sender account number"
+            "description" : "Sender bank account number to which money will be withdrawn.",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2083]: 1. Sender's bank account number from which money will be withdrawn.

File: openapi/paymentApi.yaml [hunk 5]
```diff
@@ -2021,67 +2021,91 @@
...
           "receiverName" : {
             "type" : "string",
-            "description" : "Receiver name"
+            "description" : "Receiver bank account number to which money will be withdrawn.",
+            "example" : "xyz"
           },
           "senderName" : {
             "type" : "string",
-            "description" : "Sender name associated with the payment."
+            "description" : "Sender name associated with the payment.",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2078]: Name of the sender linked to the payment.

File: openapi/paymentApi.yaml [hunk 6]
```diff
@@ -2021,67 +2021,91 @@
...
+          "forexContractNo" : {
             "type" : "string",
-            "description" : "Payment type"
+            "description" : "FX contract number is present only for the internal payment",
+            "example" : "13896558"
           },
           "receiverAccountNo" : {
             "type" : "string",
-            "description" : "Receiver account number"
+            "description" : "Receiver bank account number to which money will be deposited.",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2068]: Receiver's bank account number for the withdrawal of funds.

File: openapi/paymentApi.yaml [hunk 7]
```diff
@@ -2021,67 +2021,91 @@
...
+          "paymentAmount" : {
+            "type" : "number",
+            "description" : "Payment amount.",
+            "example" : 120.0
           },
-          "paymentType" : {
+          "forexContractNo" : {
             "type" : "string",
-            "description" : "Payment type"
+            "description" : "FX contract number is present only for the internal payment",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2063]: 1. The FX contract number is only required for internal payments.

File: openapi/paymentApi.yaml [hunk 8]
```diff
@@ -2021,67 +2021,91 @@
...
+          },
+          "wireType" : {
+            "type" : "string",
+            "description" : "Payment wire type e.g. OUTGOING_PAYMENT, INCOMING_PAYMENT",
+            "example" : "OUTGOING_PAYMENT"
           },
           "page" : {
             "type" : "integer",
-            "description" : "Which page of results are to be returned",
+            "description" : "Which page of results are to be returned. The minimum page size is 1.",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2103]: 1. What page of results should be returned? The minimum page size is 1.

File: openapi/paymentApi.yaml [hunk 9]
```diff
@@ -2021,67 +2021,91 @@
...
             "example" : 1
           },
           "limit" : {
             "type" : "integer",
-            "description" : "How many records are returned in the result set",
+            "description" : "How many records are returned in the result set. The minimum limit is 1 and the maximum limit is configurable in the system. If it is not configured then the default max limit is 1000",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2108]: Q: Where is this configuration exist? Is APIs are using this configuration? 

Number of records to be included in each page. The minimum limit is 1 and the maximum limit is configurable in the system. If it is not configured then the default max limit is 10.

File: openapi/paymentApi.yaml [hunk 10]
```diff
@@ -2522,6 +2546,12 @@
           "swiftNote" : {
             "type" : "string",
             "description" : "This is applicable for the CTP fedwire payment where the local instrument set to COVS"
+          },
+          "correspondentBankId" : {
+            "maxLength" : 10,
+            "type" : "string",
+            "description" : "This is an optional field. This is used to provide the \"routing number\" for the receiver correspondent bank in order to process the international Fedwire payment [where the receiver bank country is outside of the USA]. If the \"correspondentBankId\" is blank for the international Fedwire payment, then the Finzly system uses the internal configuration to get the correspondent bank details based on the country and currency setup.",
+            "example" : "026009593 etc."
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:2554]: Remove etc

File: openapi/paymentApi.yaml [hunk 11]
```diff
@@ -3866,52 +3896,62 @@
           "senderAddress1" : {
             "maxLength" : 35,
             "type" : "string",
-            "description" : "Sender address 1"
+            "description" : "Sender address 1",
+            "example" : "2815 Coliseum Centre Dr, Charlotte, NC 28217"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:3900]: 1. Remove , Charlotte, NC 28217

File: openapi/paymentApi.yaml [hunk 12]
```diff
@@ -3866,52 +3896,62 @@
...
           },
           "senderAddress2" : {
             "maxLength" : 35,
             "type" : "string",
-            "description" : "Sender address 2"
+            "description" : "Sender address 2",
+            "example" : "2815 Coliseum Centre Dr, Charlotte, NC 28217"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:3906]: 1. Remove , Charlotte, NC 28217

File: openapi/paymentApi.yaml [hunk 13]
```diff
@@ -3866,52 +3896,62 @@
...
-            "description" : "Sender State"
+            "description" : "Sender State",
+            "example" : "Charlotte"
           },
           "senderZip" : {
             "maxLength" : 35,
             "type" : "string",
-            "description" : "Sender Zip Code"
+            "description" : "Sender Zip Code",
+            "example" : "121254"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:3924]: 1. Update the zip as 28217

File: openapi/paymentApi.yaml [hunk 14]
```diff
@@ -3931,142 +3972,170 @@
...
-            "description" : "Sender Bank Country Name"
+            "description" : "Sender Bank Country Name",
+            "example" : "UNITED STATES OF AMERICA"
           },
           "intermediarySenderName" : {
             "maxLength" : 25,
             "type" : "string",
-            "description" : "Sender Intermediary Bank Name"
+            "description" : "Sender Intermediary Bank Name",
+            "example" : "Sender Intermediary Bank Name"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:4000]: 1. Provide valid example

File: openapi/paymentApi.yaml [hunk 15]
```diff
@@ -3931,142 +3972,170 @@
...
-            "description" : "Sender Intermediary Bank ID Type"
+            "description" : "Sender Intermediary Bank ID Type",
+            "example" : "NID"
           },
           "ultimateSenderName" : {
             "maxLength" : 255,
             "type" : "string",
-            "description" : "Ultimate Sender Name"
+            "description" : "Ultimate Sender Name",
+            "example" : "Ultimate Sender Name"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:4054]: 1. Provide valid example

File: openapi/paymentApi.yaml [hunk 16]
```diff
@@ -3931,142 +3972,170 @@
...
           },
           "ultimateSenderAccountNumber" : {
             "maxLength" : 35,
             "type" : "string",
-            "description" : "Ultimate Sender Account Number"
+            "description" : "Ultimate Sender Account Number",
+            "example" : "2265659"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:4060]: 1. Provide valid example

File: openapi/paymentApi.yaml [hunk 17]
```diff
@@ -3931,142 +3972,170 @@
...
           },
           "ultimateSenderAddress1" : {
             "maxLength" : 35,
             "type" : "string",
-            "description" : "Ultimate Sender Address 1"
+            "description" : "Ultimate Sender Address 1",
+            "example" : "Ultimate Sender Address 1"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:4066]: 1. Provide valid example

File: openapi/paymentApi.yaml [hunk 18]
```diff
@@ -3931,142 +3972,170 @@
...
           },
           "ultimateSenderAddress2" : {
             "maxLength" : 35,
             "type" : "string",
-            "description" : "Ultimate Sender Address 2"
+            "description" : "Ultimate Sender Address 2",
+            "example" : "Ultimate Sender Address 2"
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:4072]: 1. Provide valid example

File: openapi/paymentApi.yaml [hunk 19]
```diff
@@ -3931,142 +3972,170 @@
...
             "type" : "string",
-            "description" : "FBO account number"
+            "description" : "FBO account number",
+            "example" : "25896565"
           },
           "fboAccountType" : {
             "maxLength" : 30,
             "type" : "string",
-            "description" : "FBO account type"
+            "description" : "FBO account type",
```
Comment (sajeeshrosestanly) [inline openapi/paymentApi.yaml:4119]: What are the possible account types? please provide them on the description

**Review by sajeeshrosestanly — CHANGES_REQUESTED** [review]

File: openapi/paymentApi.yaml (hunk 3 above)
Comment (amruta-09) [reply openapi/paymentApi.yaml:2056]: paymentAmount is the exact amount match  and  minAmount and maxAmount if for rage

File: openapi/paymentApi.yaml (hunk 7 above)
Comment (amruta-09) [reply openapi/paymentApi.yaml:2063]: this is the search payment response

**Review by sajeeshrosestanly — APPROVED** [review]

**Pull request merged**